tensorflow/core/common_runtime/gpu/gpu_device.cc

```diff
@@ -1767,9 +1767,8 @@ Status BaseGPUDeviceFactory::GetValidDeviceIds(
                     "if you would like to use GPU. Follow the guide at "
                     "https://www.tensorflow.org/install/gpu for how to "
                     "download and setup the required libraries for your "
-                    "platform.\nSkipping registering "
-                    "GPU devices...";
-    return Status::OK();
+                    "platform.\n";
+    return errors::FailedPrecondition("Cannot dlopen some GPU libraries.");
```

Comment:
Why not return the failed `Status` in `handle_or`?

Reply:
thanks, updated.